linux/scripts/upload-to-debian.sh

```diff
@@ -73,56 +73,73 @@ if ! git diff --quiet; then
     builder_die "You have changed files in your git working directory. Exiting."
 fi
 
-builder_heading "Fetching latest changes"
-git fetch -p origin
-stable_branch=$(git branch -r | grep origin/stable- | sort | tail -1)
-stable_branch=${stable_branch##* }
-# Checkout stable branch so that `scripts/debian.sh` picks up correct version
-git checkout "${stable_branch#origin/}"
-git pull origin "${stable_branch#origin/}"
-
-if git branch -r | grep origin/beta; then
+if [[ "$(cat TIER.md)" == "beta" ]]; then
     ISBETA=true
 else
     ISBETA=false
 fi
 
+builder_heading "Fetching latest changes"
+git fetch -p origin
+if ${ISBETA}; then
+  git checkout beta
+  git pull origin beta
+else
+  stable_branch=$(git branch -r | grep origin/stable- | sort | tail -1)
+  stable_branch=${stable_branch##* }
```

Comment:
We should probably fail the script here if we don't match a stable-x.y name. I'm not entirely clear how you are deriving the stable version here

Reply:
done

Reply:
Note: will stable-100.0 sort after stable-90.0? Not a problem today but one day it may be!

Reply:
alas, I ran a quick test and I got stable-90 instead of stable-100

Reply:
This works, if we really care:

```
sort -n -k 1.8
```

* `-n` = numeric
* `-k 1.8` = sort on field 1, char position 8